File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {

    int[] stairs;

    public int climbStairs(int n) {

        this.stairs = new int[n+1];
        Arrays.fill(this.stairs, -1);

        return recursive(n);
    }

    public int recursive(int n){
        if(n <= 2)
            this.stairs[n] = n;

        if(this.stairs[n] != -1)
            return this.stairs[n];

        this.stairs[n] = recursive(n-1) + recursive(n-2);

        return this.stairs[n];
    }
}

//n=1, n = 2

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:

        dp = [ 0 for _ in range(n+1) ]

        for i in range(n+1):
            if i <= 2:
                dp[i] = i
            else:
                dp[i] = dp[i-1] + dp[i-2]

        return dp[n]